src/app/lib/atlasmap-data-mapper/services/mapping.serializer.ts

```diff
@@ -41,6 +41,7 @@ export class MappingSerializer {
                     jsonMapping = {
                         "jsonType" : "io.atlasmap.v2.Mapping",
                         "mappingType" : "MAP",
+                        'id' : m.uuid,
```

Comment:
Although it is a common practice in languages such as Java or C# when implementing iterables or lambda expressions, naming things with a single letter is discouraged in dynamically compiled languages such as JavaScript. The reason is no other than to ease type inference at a glance. Bear in mind that we currently use TypeScript, but in ES6 we do not have static type checking and figuring out types is hard.

How about repurposing the `for...of` loop above into:

```for (let mapping of mappingDefinition.mappings) {```

and then rename all the underlying references to the new name. This way any other developer will infer at a glance where are we getting that `uuid` from just reading the code. ;)

Reply:
Also, consistency is key when naming things in loosely-typed languages. The parent `for...of` could be repurposed into:

```for (let fieldMapping of mapping.fieldMappings) { ...```

Again, the index name will inform us of the underlying type of that object. ;)

Reply:
Also, such naming helps us understand that `mapping.fieldMappings` and `json.AtlasMapping.mappings.mapping` allocate the same typed collection so to say, so it's easier to undertand the different existing types at a glance. :)

Reply:
OK I think I cleaned up those not only around the fix, but entire files. One thing to note, the `mapping.fieldMappings` where `mapping` is an instance of `MappingModel`, is actually an instance of `FieldMappingPair`, which is yet another container held in the `MappingModel`. So I used `fieldMappingPair` as an instance name instead.